spec/index.md

```diff
@@ -670,37 +680,122 @@ See [The system state tree](#state-tree) for details on the state tree.
+:::
+
+Unless `requested_signatures` is null or set to `"none"` in the request and unless the query call is rejected with the reject code `SYS_TRANSIENT` (2), the response to a signed query call contains a non-empty list of signatures for the returned response produced by the individual IC nodes that computed the same returned response. Every such signature (whose type is denoted as `node-signature`) is a CBOR (see [CBOR](#cbor)) map with the following fields:
+
+-   `timestamp` (`nat`): the timestamp of the signature.
+
+-   `signature` (`blob`): the actual signature.
+
+-   `identity` (`principal`): the principal of the node producing the signature.
+
+Unless `requested_certificate` is null or set to `"no"` in the request and unless the query call is rejected with the reject code `SYS_TRANSIENT` (2), the response to a signed query call contains a certificate containing all paths in the state tree with prefixes
+
+-   `/time`. Always contained in the certificate.
+
+-   `/subnet/<subnet_id>/canister_ranges`. Contained in the certificate if the certificate does not have a subnet delegation, as specified in [Delegation](#certification-delegation).
+
+-   `/subnet/<subnet_id>/nodes/<node_id>/public_key`. Contained in the certificate if `<node_id>` is the principal of a node in `signatures`.
+
+where `<subnet_id>` characterizes the subnet to which the requested canister belongs.
+
+Given a query (the `content` map from the request body) `Q`, a response `R`, and a certificate `Cert` that is either
+
+-   contained in the response, i.e., `Cert = R.certificate`, or
+
+-   obtained by requesting the path `/subnet` in a **separate** read state request to `/api/v2/canister/<effective_canister_id>/read_state`,
+
+the following predicate describes when the returned response `R` is correctly signed by the nodes:
+
+    verify_response(Q, R, Cert)
+      = (Q.requested_signatures = null ∨ Q.requested_signatures = "none" ∨ R.reject_code = SYS_TRANSIENT => R.signatures = null) ∧
+        (Q.requested_signatures = "one" ∧ R.reject_code ≠ SYS_TRANSIENT => |R.signatures| = 1) ∧
+        (Q.requested_certificate = null ∨ Q.requested_certificate = "no" ∨ R.reject_code = SYS_TRANSIENT => Cert = null) ∧
+        verify_cert(Cert) ∧
+        ((Cert.delegation = NoDelegation ∧ SubnetId = RootSubnetId ∧ lookup(["subnet",SubnetId,"canister_ranges"], Cert) = Found Ranges) ∨
+         (SubnetId = Cert.delegation.subnet_id ∧ lookup(["subnet",SubnetId,"canister_ranges"], Cert.delegation.certificate) = Found Ranges)) ∧
+        effective_canister_id ∈ Ranges ∧
+        ∀ {timestamp: T, signature: Sig, identity: NodeId} ∈ R.signatures.
+          lookup(["subnet",SubnetId,"nodes",NodeId,"public_key"], Cert) = Found PK ∧
+          if R.status = "replied" then
+            verify_sig PK Sig ("\x0Bic-response" · hash_of_map({
```

Comment:
@mraszyk a minor thing I think this should be
```suggestion
            verify_signature PK signature ("\x0Bic-response" · hash_of_map({
```
correct? 

Same for line 790

Reply:
Thanks for the catch! Fixed.